store/db.go

```
@@ -39,44 +67,42 @@ func NewDB(dbdir string) (*DB, error) {
 	if err := os.MkdirAll(dbdir, defaultPathPerm); err != nil {
 		return nil, err
 	}
-	return &DB{dbdir: dbdir}, nil
+
+	dl, err := newDBLock(dbdir)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DB{dbdir: dbdir, dl: dl}, nil
 }
 
 func (db *DB) Open() error {
-	// take a lock on db dir
-	if db.lock != nil {
-		panic("cas db lock already gained")
-	}
-	dl, err := lock.ExclusiveLock(db.dbdir, lock.Dir)
-	if err != nil {
+	if err := db.dl.lock(); err != nil {
 		return err
 	}
-	db.lock = dl
 
 	sqldb, err := sql.Open("ql", filepath.Join(db.dbdir, DbFilename))
 	if err != nil {
-		dl.Close()
+		db.dl.unlock()
```

Ignored error here. Perhaps we should log it?